lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -341,6 +342,37 @@ module.exports = {
           const normalizedFunctionName = functionName[0].toUpperCase()
             + functionName.substr(1);
 
+          // check if LAMBDA or LAMBDA-PROXY was used for the integration type
+          if (typeof event.http === 'object') {
+            if (Boolean(event.http.integration) === true) {
+              // normalize the integration for further processing
+              const normalizedIntegration = event.http.integration.toUpperCase();
+              // check if the user has entered a non-valid integration
+              const allowedIntegrations = [
+                'LAMBDA', 'LAMBDA-PROXY',
+              ];
+              if (allowedIntegrations.indexOf(normalizedIntegration) === -1) {
+                const errorMessage = [
+                  `Invalid APIG integration "${event.http.integration}"`,
+                  ` in function "${functionName}".`,
+                  ' Supported integrations are: lambda, lambda-proxy.',
+                ].join('');
+                throw new this.serverless.classes.Error(errorMessage);
+              }
+              // map the Serverless integration to the corresponding CloudFormation types
+              // LAMBDA --> AWS
+              // LAMBDA-PROXY --> AWS_PROXY
+              if (normalizedIntegration === 'LAMBDA') {
+                integrationType = 'AWS';
+              } else if (normalizedIntegration === 'LAMBDA-PROXY') {
+                integrationType = 'AWS_PROXY';
+              } else {
+                // default to AWS_PROXY (just in case...)
+                integrationType = 'AWS_PROXY';
```

Comment:
The mapping is used so that users are not confused. In the console AWS calls it `Lambda Proxy` and `Lambda` whereas in CF they call it `AWS` and `AWS_PROXY`.

/cc @flomotlik came up with this mapping which might be confusing when you come from CF but makes sense if you played around with it on the console / lambda in general.